### PR TITLE
Replace UserCluster Prometheus with Grafana Agent

### LINKED ISSUE #11383: Deploy grafana-agent instead of prometheus in user-clusters
Use grafana-agent as a replacement for prometheus instances in user clusters for MLA purposes.

This task includes converting the configuration of prometheus to grafana-agent and replacing the deployments.

Part of implementation of #10971

### PR DESCRIPTION
**Pull request by wurbanski**
**What this PR does / why we need it**:
This replaces Prometheus inside the user cluster with grafana-agent running in metrics mode.
Removes references to "Prometheus" and replaces with generic "monitoring agent" as a counterpart to "logging agent".

**Which issue(s) this PR fixes**:

Fixes #11383 

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
UserCluster MLA: grafana-agent is now used instead of Prometheus inside the user clusters. Custom rules ConfigMaps should now be prefixed with `monitoring-scraping-` instead of `prometheus-scraping-`.
```

**Documentation**:

```documentation
https://github.com/kubermatic/docs/pull/1268
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	HAClusterIdentifier string
+}
+
+func ConfigMapCreator(config Config) reconciling.NamedConfigMapCreatorGetter {
+	return func() (string, reconciling.ConfigMapCreator) {
+		return resources.UserClusterMonitoringAgentConfigMapName, func(configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			if configMap.Data == nil {
+				configMap.Data = map[string]string{}
+			}
+			t, err := template.New("prometheus").Funcs(sprig.TxtFuncMap()).Parse(configTemplate)
```
Comment (rastislavs) [inline pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go]: What about changing the template name (`"prometheus"`) and filename in the configMap data (`"prometheus.yaml"`) to something non-prometheus as well?

Comment (wurbanski) [reply pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go]: absolutely, missed it!

**Review by xrstf — CHANGES_REQUESTED** [review]
The ticket you referenced is entirely empty. What's up with that?

Also can you please explain _why_ we are changing from Prometheus to something else? How will this affect existing KKP setups? What were the issues with Prometheus?

File: pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
```
Comment (xrstf) [inline pkg/controller/user-cluster-controller-manager/resources/resources/mla/monitoring-agent/configmap.go]: 2022

**Pull request merged**